solidity/contracts/isms/multisig/AbstractMultisigIsm.sol

```diff
@@ -57,6 +52,38 @@ abstract contract AbstractMultisigIsm is IMultisigIsm {
         uint256 _index
     ) internal pure virtual returns (bytes calldata);
 
+    /**
+     * @notice Returns the number of signatures in the metadata.
+     * @param _metadata ABI encoded module metadata
+     * @return count The number of signatures
+     */
+    function signatureCount(
+        bytes calldata _metadata
+    ) public pure virtual returns (uint256);
+}
+
+/**
+ * @title AbstractMultisigIsm
+ * @notice Manages per-domain m-of-n Validator sets of AbstractMultisig that are used to verify
+ * interchain messages.
+ */
+abstract contract AbstractMultisigIsm is AbstractMultisig {
```

Comment:
should we make an `AbstractWeightedMultisigIsm`?

Reply:
That's the non-static one.